back/spec/acceptance/resident_native_survey_responses_spec.rb

```diff
@@ -386,6 +388,7 @@
 
             # Verify that the custom field value is still the existing referenced file.
             expect(input.reload.custom_field_values).to eq({
+              'custom_field_name1' => 'Cat',
               'custom_field_name2' => { 'id' => existing_file.id, 'name' => 'existing_file.pdf' }
             })
```

Comment:
@jamesspeake Following [this commit](url), I found that this test was failing with:
```
Failure/Error:
       expect(input.reload.custom_field_values).to eq({
         'custom_field_name2' => { 'id' => existing_file.id, 'name' => 'existing_file.pdf' }
       })
     
       expected: {"custom_field_name2"=>{"id"=>"5ec099a2-135c-4c3c-8174-1c3e5c771a95", "name"=>"existing_file.pdf"}}
            got: {"custom_field_name1"=>"Cat", "custom_field_name2"=>{"id"=>"5ec099a2-135c-4c3c-8174-1c3e5c771a95", "name"=>"existing_file.pdf"}}
 ```
 In other words, before I changed this test, it seems it was overwriting the `Idea.custom_field _values` with the new key/value provided on update, whereas now it is inserting the new key/value alongside the existing key/value.
 
I have made the test pass by changing the expectation to test the value for the specific key, rather than the entire value of `custom_field_values`, because the value that was previously overwritten now still exists.
 
 Is this OK, or have I broken something here?
 Either result seems to satisfy the intention expressed by the comment
 ```
 # Verify that the custom field value is still the existing referenced file.
 ```
 and I can' think of a reason why this behaviour would be problematic. WDYT?

Reply:
If this is for a patch, then I guess it is OK, but have you changed the way updates work then to add new values instead of just patching the full set of values? In reality it'll probably make no difference given that the front end will always patch the full set of custom field values, but guess this way it will preserve custom field values in the case where a form field has changed. Don't think there should be any issue here

Reply:
That's why I was asking. I didn't think I had changed the way we patch the values, but it seems I have. In any case, it seems we agree this should not have any negative effects (it might even be slightly more robust).

Reply:
> If this is for a patch, then I guess it is OK
Yes, both tests I highlighted are for PATCH.

> but have you changed the way updates work then to add new values instead of just patching the full set of values?
That's why I was asking. I didn't think I had changed the way we patch the values, but it seems I have.
In any case, it seems we agree this should not have any negative effects (it might even be slightly more robust).